js/main.ts

```diff
@@ -17,15 +16,9 @@ let entityManager: EntityManager | null = null;
 function initialize(wasmOverrideUrl: string | null = null): Promise<void> {
   return new Promise((res, rej) => {
     if (!wasmQueue) {
-      wasmQueue = createWasmQueue(wasmOverrideUrl);
-
-      wasmQueue.runAfterWasmLoaded(() => {
-        res();
-      });
-
-      wasmQueue.runOnWasmLoadFailure(e => {
-        rej(e);
-      });
+      wasmQueue = new WasmQueue(wasmOverrideUrl)
+        .runAfterWasmLoaded(res)
+        .runOnWasmLoadFailure(rej);
```

Comment:
- [ ] Test that this doesn't change behavior. Namely, make sure that the error is still getting passed through correctly